### PR TITLE
Update to support GNOME 45

### PR DESCRIPTION
**Pull request by kenvandine**

### CODE REVIEW

**Comment by sergio-costas** [comment]
@kenvandine What write problem are you having with fprintd? I built the snap without problem, and it installed correctly...

(maybe you had, like me, a squashfs folder in the core-base-desktop folder? It made the build to fail, but I deleted it and it worked fine).

**Comment by kenvandine** [comment]
No, at runtime the service fails to start because it can't write its state.  IIRC it's under /var/lib

**Comment by sergio-costas** [comment]
You are right: from the NEWS file

```
- Create the fingerprint storage directory at install time,
  the storage path is now hard-coded as /var/lib/fprint and created by
  systemd when the service is started
```

**Comment by sergio-costas** [comment]
But it seems that it can be changed using the STATE_DIRECTORY environment variable:

imagen

File: static/etc/system-image/writable-paths [hunk 1]
```diff
@@ -69,3 +69,4 @@
 /var/lib/gdm3                           auto                    persistent  transition  none
 /usr/share/wayland-sessions             auto                    persistent  transition  none
 /usr/share/polkit-1                     auto                    persistent  transition  none
+/var/lib/colord                         auto                    persistent  transition  none
```
Comment (sergio-costas) [inline static/etc/system-image/writable-paths:72]: /var/lib/fprintd...?

Comment (kenvandine) [reply static/etc/system-image/writable-paths:72]: I forgot to push, James had added the path for colord as well

**Review by sergio-costas — APPROVED** [review]
LGTM

File: hooks/000-provide-uids-gids.chroot [hunk 2]
```diff
@@ -124,6 +124,7 @@ colord:x:113:123:colord colour management daemon:/var/lib/colord:/usr/sbin/nolog
 gnome-initial-setup:x:114:65534::/run/gnome-initial-setup/:/bin/false
 rtkit:x:115:124:RealtimeKit,,,:/proc:/usr/sbin/nologin
 pipewire:x:116:125:Pipewire,,,:/proc:/usr/sbin/nologin
+polkitd:x:999:999:polkit:/nonexistent:/usr/sbin/nologin
```
Comment (jhenstridge) [inline hooks/000-provide-uids-gids.chroot:127]: We've been following the existing practice of allocating sequential IDs for system users and groups. Is there any reason to put this up at 999?

**Review by jhenstridge — COMMENTED** [review]
This looks pretty good overall. I've added one small note about the choice of polkitd uid/gid, but other than that it seems good.

I've also created this MR to hopefully clean up the automated review failures this branch causes:

https://code.launchpad.net/~jamesh/review-tools/+git/review-tools/+merge/454104

File: hooks/000-provide-uids-gids.chroot (hunk 2 above)
Comment (kenvandine) [reply hooks/000-provide-uids-gids.chroot:127]: That wasn't intentional, I've fixed it

**Pull request merged**